tests/arch/arm/arm_thread_swap/src/arm_thread_arch.c

```diff
@@ -0,0 +1,457 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <arch/cpu.h>
+#include <arch/arm/cortex_m/cmsis.h>
+#include <kernel_structs.h>
+#include <offsets_short_arch.h>
+
+
+#if !defined(__GNUC__)
+#error __FILE__ goes only with Cortex-M GCC
+#endif
+
+#define PRIORITY 0
+#define BASEPRI_MODIFIED_1 0x20
+#define BASEPRI_MODIFIED_2 0x40
+#define SWAP_RETVAL        0x1234
+
+extern int __swap(unsigned int key);
+extern void z_move_thread_to_end_of_prio_q(struct k_thread *thread);
+
+static struct k_thread alt_thread;
+static K_THREAD_STACK_DEFINE(alt_thread_stack, 1024);
+
+int volatile switch_flag;
```

Comment:
Change this into a bool, and use "set" and "unset" terminology instead of increment when talking about it.

Otherwise we are communicating that switch_flag could have more values than 0 and 1.

Reply:
Will do, thanks.

Reply:
Done